internal/mesh/internal/controllers/gatewayproxy/builder/builder.go

```diff
@@ -192,6 +184,19 @@ func (b *proxyStateTemplateBuilder) clusters() map[string]*pbproxystate.Cluster
 		}
 	}
 
+	// Add black hole cluster for any unmatched traffic
+	clusters[xdscommon.BlackHoleClusterName] = &pbproxystate.Cluster{
```

Comment:
I'm not sure if it makes sense to share them in the longer term since our preferred behavior in the longer term for these kinds of cases will likely be aligned with the KubeSIG Gateway API contract where the rest of Consul may or may not be. Do you feel strongly about breaking this out for re-use at the moment?

Reply:
Ah right these are going to be mutually exclusive so there wouldn't be sharing per-se.

For the record here's the definition of the null route cluster today:
```
func (b *Builder) addNullRouteCluster() *Builder {
	cluster := &pbproxystate.Cluster{
		Name: NullRouteClusterName,
		Group: &pbproxystate.Cluster_EndpointGroup{
			EndpointGroup: &pbproxystate.EndpointGroup{
				Group: &pbproxystate.EndpointGroup_Static{
					Static: &pbproxystate.StaticEndpointGroup{
						Config: &pbproxystate.StaticEndpointGroupConfig{
							ConnectTimeout: durationpb.New(10 * time.Second),
						},
					},
				},
			},
		},
		Protocol: pbproxystate.Protocol_PROTOCOL_TCP,
	}

	b.proxyStateTemplate.ProxyState.Clusters[cluster.Name] = cluster
	return b
}
```

Subtly different, but possibly in a way that you would prefer to steal?